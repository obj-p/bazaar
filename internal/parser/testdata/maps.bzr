package maps

data MapFields {
    empty {string: int} = {:}
    singleValue {string: int} = {"key": 42}
    trailingComma {string: int} = {"key": 42,}
    optionalValue {string: int?} = {"key": null}
    optionalMap {string: int}?
}

func Maps() {
    var map = {:}
    var map = {a: 1}
    var map = {a: 1, b: 2,}
    var map = {"a": 1, "b": 2}
}
