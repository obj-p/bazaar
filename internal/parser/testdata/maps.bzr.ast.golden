&parser.Bazaar{
  Package: &parser.PackageDecl{
    Domain: []string{
      "maps",
    },
  },
  Decls: []*parser.TopLevelDecl{
    {
      Data: &parser.DataDecl{
        Name: "MapFields",
        Members: []*parser.MemberDecl{
          {
            Field: &parser.FieldDecl{
              Name: "empty",
              Type: &parser.TypeDecl{
                Map: &parser.MapTypeDecl{
                  Key: parser.TypeDecl{
                    Value: &"string",
                  },
                  Value: parser.TypeDecl{
                    Value: &"int",
                  },
                },
              },
              Default: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Map: &parser.MapLiteral{
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  },
}
